// Copyright 2020 The Monyhar Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.monyhar.net;

import android.support.test.InstrumentationRegistry;

import androidx.test.filters.MediumTest;

import org.junit.Test;
import org.junit.runner.RunWith;

import org.monyhar.base.test.BaseJUnit4ClassRunner;
import org.monyhar.net.test.EmbeddedTestServer;
import org.monyhar.net.test.EmbeddedTestServer.EmbeddedTestServerFailure;

/**
 * Tests for {@link EmbeddedTestServer}.
 */
@RunWith(BaseJUnit4ClassRunner.class)
public class EmbeddedTestServerTest {
    /**
     * Calling {@link EmbeddedTestServer#stopAndDestroyServer} more than once should hard fail.
     */
    @Test(expected = EmbeddedTestServerFailure.class)
    @MediumTest
    public void testServiceAliveAfterNativePage() {
        EmbeddedTestServer testServer =
                EmbeddedTestServer.createAndStartServer(InstrumentationRegistry.getContext());
        testServer.stopAndDestroyServer();
        testServer.stopAndDestroyServer();
    }
}
